### PR TITLE
makes the tram more powerful than you can ever imagine

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
tram is TOO WEAK, this pr STRENGTHENS IT. it now fires every tick instead of every decisecond, making it twice as fast and actually useful for getting around tramstation

## Why It's Good For The Game
tram is fucking RUNNING speed currently, now theres a reason to use it

## Changelog
:cl:
balance: tram is now twice as fast, pray it doesnt get faster
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/processing/tramprocess.dm [hunk 1]
```diff
@@ -2,4 +2,5 @@ PROCESSING_SUBSYSTEM_DEF(tramprocess)
 	name = "Tram Process"
 	wait = 1
 	/// only used on maps with trams, so only enabled by such.
```
Comment (Thunder12345) [inline code/controllers/subsystem/processing/tramprocess.dm:4]: This should probably stay where it was

File: code/controllers/subsystem/processing/tramprocess.dm [hunk 2]
```diff
@@ -2,4 +2,5 @@ PROCESSING_SUBSYSTEM_DEF(tramprocess)
...
+	flags = SS_TICKER
 	can_fire = FALSE
```
Comment (Fikou) [inline code/controllers/subsystem/processing/tramprocess.dm:6]: move this up so the comment makes sense

File: code/controllers/subsystem/processing/tramprocess.dm (hunk 1 above)
Comment (Fikou) [inline code/controllers/subsystem/processing/tramprocess.dm:4]: this shouldnt be an autodoc comment

**Comment by LemonInTheDark** [comment]
Muh atmos system

**Comment by Kylerace** [comment]
im a stupid dumb idiot who should eat before opening but tram will be faster

**Comment by moocowswag** [comment]
This would warrant a need for there to be some sort of stoplight at the tram crosswalk when the tram is coming.

File: code/controllers/subsystem/processing/tramprocess.dm (hunk 2 above)
Comment (Kylerace) [reply code/controllers/subsystem/processing/tramprocess.dm:6]: done

File: code/controllers/subsystem/processing/tramprocess.dm (hunk 1 above)
Comment (Kylerace) [reply code/controllers/subsystem/processing/tramprocess.dm]: done

Comment (Kylerace) [reply code/controllers/subsystem/processing/tramprocess.dm]: yes

**Comment by AlinaStarkova** [comment]
> 
> 
> This would warrant a need for there to be some sort of stoplight at the tram crosswalk when the tram is coming.

yeah, tram is already a risk of getting hit when you cross, if it can now hit you without any way to dodge once you see it enter the screen due to its speed being too fast then its going to be very deadly

also with tram being not any faster than running it also helped push people down below using the ladder accesses which helped with antags being able to catch people alone. currently theres no reason to use maints, as maints is lower deck with almost nothing of signifigance down there, not to mention upper deck can see down a large portion of it

**Comment by MMMiracles** [comment]
anyone who is arguing against a faster tram is a heretic

faster TRAM FASTER TRAM

**Comment by coiax** [comment]
If you don't want to get hit by the tram, don't walk on the tram tracks.

**Comment by MMMiracles** [comment]
if this gets merged I'll add a lights system since the tram will actually be fast enough to warrant some cautionary signs

**Comment by MrDoomBringer** [comment]
i tried this change out locally, good lord it's fun. 👍 👍

**Comment by Mothblocks** [comment]
> This would warrant a need for there to be some sort of stoplight at the tram crosswalk when the tram is coming.

the whole point is you take the tram bro

**Review by Mothblocks — APPROVED** [review]
idk why this is drafted

**Comment by interestingusernam3** [comment]
🚀 tram 🚀 🚀

**Comment by AlinaStarkova** [comment]
> > This would warrant a need for there to be some sort of stoplight at the tram crosswalk when the tram is coming.
> 
> the whole point is you take the tram bro

your aware you have to cross the track to go north or south right? you dont take a tram just to pass by 8 tiles

**Comment by Mothblocks** [comment]
dont be surprised when it slams into you then thats the risk you take

**Comment by AlinaStarkova** [comment]
> 
> 
> dont be surprised when it slams into you then thats the risk you take

but your required to pass the tracks often, i dont think a map should be designed behind "hey you have to do X to play the map, but doing X might kill you, and thats your fault" i dont see the harm in adding lights that turn red if the tram is moving your direction, and green when the trams either stopped or moving away from that direction

**Comment by Mothblocks** [comment]
because then less people will get hit and its funny when they do

**Comment by MrDoomBringer** [comment]
@AlinaStarkova weren't you just complaining about how this would cause less people to go into maint? If you really don't wanna get hit just take the ladder down into maint and cross north/south, just like you would for crossing east/west

**Comment by AlinaStarkova** [comment]
> 
> 
> @AlinaStarkova weren't you just complaining about how this would cause less people to go into maint? If you really don't wanna get hit just take the ladder down into maint and cross north/south, just like you would for crossing east/west

two things can be true at once. while its true encouraging people to use maints is a good thing, encouraging them to go into maints because they will otherwise get their ass handed to them by a train they cant see probably isnt the way to do that imho

**Comment by Kylerace** [comment]
@Mothblocks 
>idk why this is drafted
subsystems with the SS_TICKER flag are automatically given a higher priority than non tickers, meaning if SStramprocess was too expensive it would lag out every non ticker subsystem. i could override that with SS_BACKGROUND but that would cause it to not fire when the server is laggy. right now im looking into making a version of SS_TICKER that doesnt have the priority bump

**Comment by Mothblocks** [comment]
sounds good

**Comment by MrDoomBringer** [comment]
> encouraging them to go into maints because they will otherwise get their ass handed to them by a train they cant see probably isnt the way to do that imho

Why is that a bad thing? Adding stoplights just takes away from the only interesting part of this station: a classic risk reward scenario. 
- Safest option: calling the tram and using it to cross, takes like 10 seconds given how fast the tram moves now
- Middle ground: going via maint where antags could nab you, but you're guaranteed to not be hit
- Risky: cross them tracks baby 

This is only even a consideration for the middle stop on the map, the left and right stops have absolutely 0 risk of getting hit by the train.

**Comment by ilikesaying** [comment]
nothing stops the Tram-Dozer

**Comment by MrDoomBringer** [comment]
> subsystems with the SS_TICKER flag are automatically given a higher priority than non tickers, meaning if SStramprocess was too expensive it would lag out every non ticker subsystem. i could override that with SS_BACKGROUND but that would cause it to not fire when the server is laggy. right now im looking into making a version of SS_TICKER that doesnt have the priority bump

Was this resolved?

**Comment by Mothblocks** [comment]
in testing it wasnt a real issue

**Comment by MrDoomBringer** [comment]
eery nice

**Pull request merged**